client/landing/stepper/declarative-flow/internals/components/stepper-loader/index.tsx

```diff
@@ -1,10 +1,46 @@
-import clsx from 'clsx';
-import WordPressLogo from 'calypso/components/wordpress-logo';
+import { isWooExpressFlow, isTransferringHostedSiteCreationFlow } from '@automattic/onboarding';
+import { ProgressBar } from '@wordpress/components';
+import { LoadingBar } from 'calypso/components/loading-bar';
 import './style.scss';
 
-const StepperLoader = () => {
-	/* eslint-disable wpcalypso/jsx-classname-namespace */
-	return <WordPressLogo size={ 72 } className={ clsx( 'wpcom-site__logo', 'stepper-loader' ) } />;
+interface StepperLoaderProps {
+	title?: string;
+	subtitle?: React.ReactNode;
+	progress?: number;
+	flow?: string;
+}
+
+const StepperLoader: React.FC< StepperLoaderProps > = ( {
+	title,
+	subtitle,
+	progress = -1,
+	flow = null,
+} ) => {
+	const renderProgressComponent = () => {
+		if ( isWooExpressFlow( flow ) || isTransferringHostedSiteCreationFlow( flow ) ) {
+			return (
+				<LoadingBar
+					progress={ progress >= 0 ? progress : -1 }
+					className="processing-step__content woocommerce-install__content"
+				/>
+			);
+		}
+
+		return (
+			<ProgressBar
+				value={ progress >= 0 ? progress * 100 : undefined }
+				className="stepper-loader__progress-bar processing-step__progress-bar"
+			/>
+		);
+	};
+
+	return (
+		<div className="stepper-loader">
+			<h1 className="stepper-loader__title processing-step__progress-step">{ title }</h1>
+			{ renderProgressComponent() }
+			{ subtitle && <p className="processing-step__subtitle">{ subtitle }</p> }
```

Comment:
I've lost track a bit about how we add classes, but where are these coming from and why aren't they namespaced to stepper-loader?

Reply:
That's a great question. We should change this. These come from the processing step in Stepper but it's not guaranteed to be loaded in all flows.

Reply:
Totally right. It was a pre-existing class name taken from the Processing step. It is now `stepper-loader__subtitle`